### PR TITLE
Base icon btn

### PR DESCRIPTION
**Pull request by Caleb335**
Issue: #549
- [x] Create Base Icon Button with the formatting needed to make it look good, without adding any icon to it.
- [x] Add an additional class for positioning of the icon (left or right of the text).
- [x] Test that it would work fine with and without text.
- [x] Add to the documentation (Readme and Get Started page) a note about using any icon with the base icon button.

I've added the `base-icon-btn` class and how to use-it guide in the README file and on the website. Kindly review it.

Below are some of the screenshots I took while testing the button.

**With text**
Screenshot from 2020-10-04 18-49-25

**without text**
Screenshot from 2020-10-04 19-09-10

Kindly review it, thanks

### CODE REVIEW

**Review by shahednasser — CHANGES_REQUESTED** [review]
You did a great job! I have a few remarks:

1. Can we make the icon by default in the center? if it doesn't have classes `icon-left` and `icon-right` then it should be in the center.
2. Please add in the documentation (README and website) the purpose of the base icon button which is to user any button the user might want and not just the ones listed on the website.
3. in the get started page please add "Base Icon Button" to the sidebar"
4. There seems to be a problem in the `dist/sbuttons.css` file due to merge conflict. Please run `npm run compile` again as I can't see the changes you made. Don't resolve conflict for now, we will work it out in the end.
5. Place "Base Icon Button" in the examples page as the first among "Icon Buttons"

**Comment by Caleb335** [comment]
> You did a great job! I have a few remarks:

Thank you.

> 1. Can we make the icon by default in the center? if it doesn't have classes `icon-left` and `icon-right` then it should be in the center.

Yes, we can. I'd add that style to it now.
> 2. Please add in the documentation (README and website) the purpose of the base icon button which is to user any button the user might want and not just the ones listed on the website.

I've actually added the details and the purpose of the `base icon` button in the README and on the website, I guess it was as a result of the compiling/merge conflict issue that was why you were not able to see it, but I've updated it now, you can view the deploy preview [here](https://deploy-preview-636--clever-boyd-7dd108.netlify.app/#base-icon) and the [get started](https://deploy-preview-636--clever-boyd-7dd108.netlify.app/getstarted) documentation.

> 3. in the get started page please add "Base Icon Button" to the sidebar"

Okay, I'd do that now also.

> 5. Place "Base Icon Button" in the examples page as the first among "Icon Buttons"

On it.

**Comment by Caleb335** [comment]
@shahednasser I've added all the changes.

Kindly review 🙏🏼

**Comment by shahednasser** [comment]
I've seen the details you added but it mainly just talks about the positioning of the icon. We need to add the purpose (example: to use any button you want...), then how to use it (example: include an i element for the icon inside the button), then the position of the icon which is what you wrote originally.

And make sure it's the same in the README and the Get Started page.

Also for the example can you use different icons to showcase the examples? I also suggest adding three types of buttons: 
1. left which will include text + the button
2. center without text
3. right which will include text + the button.

I forgot to mention this before but the button needs to have a transparent background with the default text color black (create a new variable for it). Then we need to add cases for example if used with `blue-btn` then the color of the text should be white and so on. Once you make the background transparent just show the base button without rounded and block.

**Comment by Caleb335** [comment]
> I've seen the details you added but it mainly just talks about the positioning of the icon. We need to add the purpose (example: to use any button you want...), then how to use it (example: include an i element for the icon inside the button), then the position of the icon which is what you wrote originally.
> 
> And make sure it's the same in the README and the Get Started page.
> 
> Also for the example can you use different icons to showcase the examples? I also suggest adding three types of buttons:
> 
> 1. left which will include text + the button
> 2. center without text
> 3. right which will include text + the button.
> 
> I forgot to mention this before but the button needs to have a transparent background with the default text color black (create a new variable for it). Then we need to add cases for example if used with `blue-btn` then the color of the text should be white and so on. Once you make the background transparent just show the base button without rounded and block.

Noted, I'd work on all this now.

**Comment by Caleb335** [comment]
> Once you make the background transparent just show the base button without rounded and block.

Hi @shahednasser, 

About this, if I apply the transparent-btn class to the base button, the button text won't be visible. 

Should I go ahead? I just want to be clear, before I go on with the implementation.

BTW, I've made the default text-color of the `base-icon-btn` to be black and I also added a border and tested other button colors with it too.

Check it out in the screenshot below. 

**DEFAULT BASE ICON BUTTON**
Screenshot from 2020-10-05 10-08-25

**WITH OTHER COLORS**
Screenshot from 2020-10-05 10-11-01

**Comment by shahednasser** [comment]
W need the base class without a border and without a background color. You can checkout for example the star button (ignore the active state). That's basically how the icon should look like using just the basic class.

But we should make the icon look good when using color buttons like `blue-btn`. 

Then after looking at it, I think the example should show the following:
1. The base class without any color buttons.
2. Showcase left, center and right using one color button as an example.
3. Showcase left, center and right using one color button as an example and it should be rounded.
4. Same as the above but block

Please let me know if you need any assistance or have any questions

**Comment by Caleb335** [comment]
> W need the base class without a border and without a background color. You can checkout for example the star button (ignore the active state). That's basically how the icon should look like using just the basic class.
> 
> But we should make the icon look good when using color buttons like `blue-btn`.
> 
> Then after looking at it, I think the example should show the following:
> 
> 1. The base class without any color buttons.
> 2. Showcase left, center and right using one color button as an example.
> 3. Showcase left, center and right using one color button as an example and it should be rounded.
> 4. Same as the above but block
> 
> Please let me know if you need any assistance or have any questions

Thank You for this elaborate breakdown, I'd implement everything you've mentioned.

When I get stuck I'd let you know I need help

Thank you.

File: README.md [hunk 1]
```diff
@@ -113,6 +113,92 @@ In case of using buttons that have icons in them, make sure to include font awes
...
+
+- ## Usage
+
+Examples
+- [Base buttons](#base-icon-button)
+- [Rounded buttons](#rounded-buttons)
+- [Block buttons](#block-icon-buttons)
+
+* ### base icon button 
+ The `base-icon-btn class`, gives you a default button with no background color and a border. You can choose to style it by adding, some of the other available classes, like `btn-orange` to change the background color, and `icon-right` to position the icon in the button. You can make use of it, by copying the sinppet below.
```
Comment (shahednasser) [inline README.md:128]: class should be outside the `

Comment (shahednasser) [reply README.md:128]: "You can choose to style it by adding some ..." remove the comma after adding

**Review by shahednasser — CHANGES_REQUESTED** [review]
You did a great job writing the documentation! I have the following notes: 

1. Add "Base Icon Buttons" to the Contents Menu in README with its submenu items (Base Buttons, rounded buttons, etc...)
2. Remove the rounded dot before every list item (the *)
3. Capitalize first letter of all headings (base icon button, left icon button, etc...) (for the get started page as well)
4. In the button example, use any icon for the example. Make the icons different to demonstrate that it works for whatever item (Note: to add the icon I suggest adding the i element in the "label" property of the button but honestly I haven't tested that. If there are any changes that need to be made to make it work then you can go ahead and make them.)
5. Remove the hover effect on the base icon button. I don't think it's necessary.
6. In the Get Started page also use random icons for the "base icon button" example.
7. Add submenus in the Get Started page as mentioned for the README

File: README.md (hunk 1 above)
Comment (Caleb335) [reply README.md:128]: Ok, thank you for noting this.

Comment (Caleb335) [reply README.md:128]: I've  resolved that

**Comment by Caleb335** [comment]
> You did a great job writing the documentation! I have the following notes:
> 
> 1. Add "Base Icon Buttons" to the Contents Menu in README with its submenu items (Base Buttons, rounded buttons, etc...)
> 2. Remove the rounded dot before every list item (the *)
> 3. Capitalize first letter of all headings (base icon button, left icon button, etc...) (for the get started page as well)
> 4. In the button example, use any icon for the example. Make the icons different to demonstrate that it works for whatever item (Note: to add the icon I suggest adding the i element in the "label" property of the button but honestly I haven't tested that. If there are any changes that need to be made to make it work then you can go ahead and make them.)
> 5. Remove the hover effect on the base icon button. I don't think it's necessary.
> 6. In the Get Started page also use random icons for the "base icon button" example.
> 7. Add submenus in the Get Started page as mentioned for the README

Wow! 👀🤯

This is alot, I'm working on it 🚀

**Comment by shahednasser** [comment]
Take your time with it and thank you for all your work!

**Comment by Caleb335** [comment]
Hi @shahednasser, I've added all the changes you requested.

Kindly review.

> 4. In the button example, use any icon for the example. Make the icons different to demonstrate that it works for whatever item (Note: to add the icon I suggest adding the i element in the "label" property of the button but honestly I haven't tested that. If there are any changes that need to be made to make it work then you can go ahead and make them.)
Remove the hover effect on the base icon button. I don't think it's necessary.

But, regarding this task above I tried placing the `i` element in the `label` property, the code breaks whenever I tried that. So here's my suggestion... I think we should leave the `base-icon-button` in the examples plain without adding any icon, so it actually serves the purpose "base button".

But if anyone comes up with a solution, it should be implemented.

**Review by shahednasser — CHANGES_REQUESTED** [review]
Hi @Caleb335 so I checked regarding the label part and I was actually the one mistaken. It should be added in `text` not `label` and it will work fine (I tested it).

Everything else looks great! The only remark I have is in "Get Started" page in the Base button example just keep the book icon without "base button"

**Comment by Caleb335** [comment]
> Hi @Caleb335 so I checked regarding the label part and I was actually the one mistaken. It should be added in `text` not `label` and it will work fine (I tested it).
> 
> Everything else looks great! The only remark I have is in "Get Started" page in the Base button example just keep the book icon without "base button"

Alright, I'd fix that.

**Comment by Caleb335** [comment]
> Hi @Caleb335 so I checked regarding the label part and I was actually the one mistaken. It should be added in `text` not `label` and it will work fine (I tested it).

Is this what you mean?...

Screenshot from 2020-10-06 09-05-24

**Comment by shahednasser** [comment]
No what I mean something like:

```
text: '<i class="fas fa-sliders-h"></i>'
```

**Comment by Caleb335** [comment]
> No what I mean something like:
> 
> ```
> text: '<i class="fas fa-sliders-h"></i>'
> ```

Nice! Thank you.

It works now.

Screenshot from 2020-10-06 09-14-55

**Comment by shahednasser** [comment]
Great but can you make those buttons colored (for example blue-btn) and add one in the middle for the default look without "base icon button" text (just the icon)?

**Comment by Caleb335** [comment]
> Great but can you make those buttons colored (for example blue-btn) and add one in the middle for the default look without "base icon button" text (just the icon)?

Ok, I'd add those changes too. ✌️

**Comment by Caleb335** [comment]
Hi @shahednasser, I've added the colors and the default button also.

What else would you like to be added before I commit the changes?
Screenshot from 2020-10-06 13-05-02

**Comment by shahednasser** [comment]
The padding in the left icon looks different that of how it looks like in the getting started page. I'm not sure if you can fix that but other than that no these are the last changes.

**Comment by Caleb335** [comment]
> The padding in the left icon looks different that of how it looks like in the getting started page. I'm not sure if you can fix that but other than that no these are the last changes.

Ok...

I'd try adjusting the padding of the left icon.

**Comment by Caleb335** [comment]
All done @shahednasser

**Comment by shahednasser** [comment]
Great work thank you! I know this was a lot of work but thank you for contributing greatly to our repository I really appreciate it.

@all-contributors please add @Caleb335 for code, content and example

**Comment by Caleb335** [comment]
> Great work thank you! I know this was a lot of work but thank you for contributing greatly to our repository I really appreciate it.
> 
> @all-contributors please add @Caleb335 for code, content and example

Thank you...

It really was a lot of work! Now I've got to go take a deep nap 😄

**Pull request merged**